lib/active_merchant/billing/gateways/omise.rb

```diff
@@ -0,0 +1,376 @@
+      #
+      # ==== Parameters
+      #
+      # * <tt>payment</tt> -- The CreditCard.
+      # * <tt>options</tt> -- A standard hash options.
+      # * use options set_default_card: true to set the default card
+
+      def store(payment, options={})
+        return Response.new(false, "#{payment.errors.full_messages.join('. ')}") if payment.is_a?(CreditCard) and !payment.valid?
+        post, card_params = {}, {}
+        add_customer_data(post, options) # add description, email params etc.
+        add_or_create_token(card_params, payment, options) # add token or card to params[:card] when it's valid.
+        if options[:customer_id] # attach the card to this customer
+          attach_customer_card(post, card_params, options)
+        else
+          commit(:post, 'customers', post.merge(card_params), options)
+        end
+      end
+
+      # Delete a customer or associated credit card.
+      #
+      # ==== Parameters
+      #
+      # * <tt>customer_id</tt> -- The Customer identifier (REQUIRED).
+      # * <tt>options</tt>     -- A standard hash options.
+
+      def unstore(customer_id, options={})
+        return unless customer_id
+        return commit(:delete, "customers/#{CGI.escape(customer_id)}") unless options[:card_id]
+        commit(:delete, "customers/#{CGI.escape(customer_id)}/cards/#{CGI.escape(options[:card_id])}")
+      end
+
+
+      # Verify a Card.
+      #
+      # ==== Parameters
+      #
+      # * <tt>payment</tt> -- The CreditCard.
+      # * <tt>options</tt> -- A standard hash options.
+
+      def verify(payment, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(25, payment, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      # Enable scrubbling sensitive information
+      def supports_scrubbing?
+        true
+      end
+
+      # Scrub sensitive information out of HTTP transcripts
+      #
+      # ==== Parameters
+      #
+      # * <tt>transcript</tt> -- The HTTP transcripts
+
+      def scrub(transcript)
+        transcript.
+          gsub(/(Authorization: Basic )\w+/i, '\1[FILTERED]').
+          gsub(/(\\"number\\":)\\"\d+\\"/, '\1[FILTERED]').
+          gsub(/(\\"security_code\\":)\\"\d+\\"/,'\1[FILTERED]')
+      end
+
+      private
+
+      def omise_tokenize(payment, options={})
+        options[:key] ||= @public_key
+        data = {}
+        add_creditcard(data, payment)
+        options[:url_endpoint] = VAULT_URL + 'tokens'
+        response = https_request(:post, nil, data, options)
+        token =
+          if successful?(response)
+            Response.new(true, nil, token: response)
+          else
+            Response.new(false, message_from(response), {}, error_code: STANDARD_ERROR_CODE_MAPPING[error_code_from(response)])
+          end
+        OmisePaymentToken.new(token)
+      end
+
+      def add_or_create_token(post, payment, options={})
```

Comment:
Can you split this into two methods, one for adding to a request, and one for creating a token? I think that'll make the code clearer.